src/cosalib/qemuvariants.py

```diff
@@ -321,7 +321,7 @@ def mutate_image(self):
                 case "gzip":
                     rc = ['gzip', '-9c', uncompressed_path]
                 case "zip":
-                    rc = ['zip', '-9', "-", uncompressed_path]
+                    rc = ['zip', '-9j', "-", uncompressed_path]
```

Comment:
This is a non-HyperV specific code path (i.e. if we added another platform in the future that wanted zip). Is this Hyper-V specific or do we think if another platform wanted zip they would also want this option?

Reply:
`uncompressed_path` is an absolute path, so we were ending up with a ZIP file containing e.g. `srv/builds/38.20230628.dev.1/x86_64/fedora-coreos-38.20230628.dev.1-hyperv.x86_64.vhdx`.  Every future platform will want those intermediate directories dropped.